src/audiomixcontroller/DefaultAudioMixController.ts

```diff
@@ -56,13 +55,13 @@ export default class DefaultAudioMixController implements AudioMixController {
         const oldSinkId: string = this.audioElement.sinkId;
         if (newSinkId !== oldSinkId) {
           if (this.browserBehavior.hasChromiumWebRTC()) {
-            new AsyncScheduler().start(async () => {
-              const existingStream = await this.audioElement.srcObject;
-              this.audioElement.srcObject = null;
-              // @ts-ignore
-              await this.audioElement.setSinkId(newSinkId);
-              this.audioElement.srcObject = existingStream;
-              this.audioStream = existingStream as MediaStream;
+            const existingStream = this.audioElement.srcObject;
+            this.audioElement.srcObject = null;
+            // @ts-ignore
+            this.audioElement.setSinkId(newSinkId).then(() => {
```

Comment:
What will happen when `setSinkId` fails?

In the current PR, `audioElement.srcObject` will be set to `null`. You can catch the error and restore `srcObject` but it'll be difficult to figure out the error as this compoent doens't have any logger.

Reply:
Yeah if we catch the exception thrown by setSinkId(), we might miss on errors generated from the setSinkId() eg the DOMException